test/integration/runIntegrationTests.ts

```diff
@@ -148,15 +158,36 @@ class TestHarness {
         output += data
       })
       proc.on('exit', (code) => {
-        if (!throwOnError || code === 0) {
-          resolve({
-            output: cleanup({ text: output, rootDir: this.config.dir }),
-            status: code ?? 1,
-          })
+        if (expectError) {
+          if (code === 0) {
+            // eslint-disable-next-line no-console
+            console.error(output)
+            reject(
+              new Error(
+                `Exited with code ${code} ${cleanup({ text: output, rootDir: this.config.dir })}`,
+              ),
+            )
+          } else {
+            resolve({
+              output: cleanup({ text: output, rootDir: this.config.dir }),
+              status: code ?? 1,
+            })
+          }
         } else {
-          // eslint-disable-next-line no-console
-          console.error(output)
-          reject(new Error(`Exited with code ${code ?? 'null'}`))
+          if (code === 0) {
+            resolve({
+              output: cleanup({ text: output, rootDir: this.config.dir }),
+              status: code ?? 1,
+            })
+          } else {
+            // eslint-disable-next-line no-console
+            console.error(output)
+            reject(
+              new Error(
+                `Exited with code ${code} ${cleanup({ text: output, rootDir: this.config.dir })}`,
+              ),
+            )
+          }
```

Comment:
this can be simplified back down to two branches like this (pseudocode)

    if ((expectError && didError) || (!expectError and !didError)) {
       return the output and status code
    } else {
       throw the error
    }

Reply:
Yeah, good point!